packages/devtools/lib/src/memory/memory.dart

```diff
@@ -486,3 +992,152 @@ class MemoryScreen extends Screen with SetStateMixin {
     }
   }
 }
+
+/// Path consists of:
+///    Class selected (from Class list):
+///      _className
+///      _hashCode = empty
+///      field = empty
+///
+///   Instance selected (from Instance list):
+///      _className
+///      _hashCode [hashCode of instance]
+///      field = empty
+///
+///   Hover (from inboundReferences) parent allocations:
+///      _className [class name of parent class that allocated object]
+///      _hashCode [hashCode of instance]
+///      field [field of parent class that has ref]
+class NavigationState {
+  NavigationState._()
+      : _className = '',
+        _hashCode = '';
+  NavigationState.classSelect(this._className) : _hashCode = '';
+  NavigationState.instanceSelect(this._className, this._hashCode);
+
+  static const String dataIndex = 'data-index';
+  static const String dataClass = 'data-class';
+  static const String dataField = 'data-field';
+  static const String dataHashCode = 'data-hashcode';
+
+  final String _className;
+  final String _hashCode;
+
+  String field = '';
+
+  String get className => _className;
+  String get objectHashCode => _hashCode;
+
+  bool get isClass =>
+      _className.isNotEmpty && field.isEmpty && _hashCode.isEmpty;
+  bool get isInstance =>
+      _className.isNotEmpty && field.isEmpty && _hashCode.isNotEmpty;
+  bool get isInbound =>
+      _className.isNotEmpty && field.isNotEmpty && _hashCode.isNotEmpty;
+
+  // Create a span with all information to navigate through the class list and
+  // instance list. The span element will look like:
+  //
+  //    <span class=N data-index=# data-class=N data-field=N data-hashcode=N>
+  //      class[.field]
+  //    </span>
+  //
+  // where:
+  //    class=N is the css class for styling
+  //    index=# is the index of this Navigation link in the NavigationPath list
+  //    data-class=N is the class selected in the memory class list
+  //    data-field=N if specified, references previous history hashcode (object)
+  //    data-hashcode=N if specified, object referenced in this data-class field
+  CoreElement link(int index, [bool last = false]) {
+    final String spanText = field.isNotEmpty
+        ? '$className.$field'
+        : isInstance ? '$className (instance)' : className;
+
+    final CoreElement spanElem =
+        span(text: spanText, c: last ? 'history-link-last' : 'history-link');
+
+    spanElem.setAttribute(dataIndex, '$index');
+    spanElem.setAttribute(dataClass, className);
+    if (field.isNotEmpty) spanElem.setAttribute(dataField, field);
+    if (objectHashCode.isNotEmpty) {
+      spanElem.setAttribute(dataHashCode, objectHashCode);
+    }
+
+    return spanElem;
+  }
+
+  CoreElement get separator => span(text: '>', c: 'history-separator');
+}
+
+class NavigationPath {
```

Comment:
could you add dartdoc to this class explaining what it is?

Reply:
Done.